리뷰요청!

hjhello423 opened the pull request:
안녕하세요 민준님 😄
setp1 진행해 보았습니다!! 리뷰 부탁드려요 🙏
잘 부탁드립니다!!!

hjhello423 committed:
feat: 인증 정보 converter 추가

hjhello423 committed:
test: 인증 정보 convert 테스트 코드 추가

hjhello423 committed:
feat: 인증 추상화 class 추가

hjhello423 committed:
refactor: 리팩토링한 interceptor 교체

hjhello423 committed:
refactor: class 이름 수정 사항 반영

hjhello423 committed:
refactor: 메서드 이름 수정

hjhello423 committed:
feat: Basic, Bearer converter 추가

hjhello423 committed:
test: Basic, Bearer converter 테스트 코드 추가

hjhello423 committed:
refactor: 인증 interceptor 추상화 리팩토링

hjhello423 committed:
refactor: 추상화 리팩토링 class 교체

hjhello423 committed:
feat: 의존성 제거 interface 추가

hjhello423 committed:
refactor: LoginMemberService의 의존성 제거

hjhello423 committed:
refactor: LoginMember에 대한 의존성 제거

starkim06 commented on src/main/java/nextstep/auth/application/LoginMemberService.java:
```diff
@@ -0,0 +1,9 @@
+
+public interface LoginMemberService {
+
+    LoginMember findByUsername(String email);
```
return type의 class는 LoginMember이고 method name에서는 User를 사용하고 있는 부분과
method name의 name과 parameter의 email 부분이 읽을때 어색한 느낌이 있는것 같아요
naming을 통일시켜보는건 어떨까요?

hjhello423 replied:
흠 그러네요
이부분은 기존에 있던 LoginMemberService를 다시 살려서 처래해보았습니다 👍

starkim06 commented on src/main/java/nextstep/auth/authentication/AuthenticationToken.java:
```diff
@@ -9,6 +9,10 @@ public AuthenticationToken(String principal, String credentials) {
         this.credentials = credentials;
     }
 
+    public static AuthenticationToken of(String principal, String credentials) {
```
추가하신 method를 사용하는 부분과 사용하지 않는 부분들이 같이 있는것 같아요!
같이 사용하고 있는 특별한 이유가 없다면 생성자와 추가하신 method 중에서 하나만 사용하는것으로 통일시켜보는건 어떨까요?

starkim06 commented on src/main/java/nextstep/member/application/MemberService.java:
```diff
@@ -46,4 +48,10 @@ public void deleteMember(Long id) {
-}
\ No newline at end of file
+
+    public LoginMember findByUsername(String email) {
```
LoginMemberService에서 method를 오버라이드 하고 있는것 같아요!
@Override annotation을 명시해주는건 어떨까요?

starkim06 commented on src/main/java/nextstep/auth/application/LoginMemberService.java:
```diff
@@ -0,0 +1,9 @@
+
+public interface LoginMemberService {
+
+    LoginMember findByUsername(String email);
```
LoginMember를 아직 사용하고 있어서 auth package에서 member package로의 의존성이 제거되지 않은것 같아요!

starkim06 commented on src/main/java/nextstep/auth/authentication/NonChainAuthenticationInterceptor.java:
```diff
@@ -0,0 +1,42 @@
+        this.authenticationConverter = authenticationConverter;
+    }
+
+    public abstract void afterAuthentication(LoginUser loginMember, HttpServletResponse response) throws Exception;
```
클래스명과 다른 loginMember 라는 변수명을 사용하시는 이유가 있을까요?

hjhello423 replied:
수정해보았습니다!

starkim06 commented on src/main/java/nextstep/auth/authentication/NonChainAuthenticationInterceptor.java:
```diff
@@ -0,0 +1,42 @@
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        AuthenticationToken authentication = authenticationConverter.convert(request);
+
+        LoginUser loginMember = loginMemberService.findByUsername(authentication.getPrincipal());
```
LoginUser를 통한 인증작업을 NonChainAuthentication 이라는 추상 클래스가 하기에는 너무 구체적(?) 이지 않나 라는 생각도 드는것 같아요.
LoginUser를 통한 인증 로직의 중복 제거를 하는거라면 NonChainAuthenticationInterceptor를 상속받는 별도의 클래스가 있으면 좋을것도 같아요.
홍준님 생각은 어떠신가요?

starkim06 commented on src/main/java/nextstep/auth/authentication/AuthenticationConverter.java:
```diff
@@ -0,0 +1,9 @@
+
+import javax.servlet.http.HttpServletRequest;
+
+public interface AuthenticationConverter {
```
AuthenticationConverter를 별도의 클래스로 분리 :100:

starkim06 commented on src/main/java/nextstep/auth/authentication/BasicAuthenticationFilter.java:
```diff
@@ -1,50 +1,35 @@
-            String authCredentials = AuthorizationExtractor.extract(request, AuthorizationType.BASIC);
-            String authHeader = new String(Base64.decodeBase64(authCredentials));
+    public void afterAuthentication(AuthenticationToken token) {
 
```
사소한거지만 method의 첫줄과 마지막 줄에 줄바꿈을 해주신 이유가 있을까요?

hjhello423 replied:
step1 진행하면 서 엉뚱한 prj에서 개발을 했었는데
위에 리뷰 해주신 몇개랑 같이 코드 옮기는 과정에서 놓친거 같네요 ㅠ

starkim06 commented on src/main/java/nextstep/auth/authentication/ChainAuthenticationInterceptor.java:
```diff
@@ -0,0 +1,31 @@
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        try {
+            AuthenticationToken token = authenticationConverter.convert(request);
+            afterAuthentication(token);
```
method name은 afterAuthentication이지만 상속받은 method들이 하는 역할은 authentication과 afterAuthentication두가지 모두를 하는것 같아요! 
홍준님 생각은 어떠신가요?

hjhello423 replied:
ChainAuthenticationInterceptor, NonChainAuthenticationInterceptor의 afterAuthentication()을 구현한 메서드들이
preHandle()에서 인증을 처리 한 후 적절한 응답을 만드는 과정이라고 생각했었어요.
그런데 BasicAuthenticationFilter의 afterAuthentication()에서는 인증의 결과로 생성된 객체를 다시 검증하고 있어서 
말씀하신 것처럼 인증을 양쪽(?)에서 처리하는걸로 보여지네요

starkim06 reviewed (commented):
안녕하세요 홍준님!
1주 차 미션 너무 잘해주신 것 같아요!!! :100:
몇 가지 코멘트 남겼습니다!
확인해 보시고 궁금한 점이나 제가 잘못 이해한 부분들이 있으면 말씀 부탁드려요!

hjhello423 committed:
refactor: 의존성 제거 수정 및 리팩토링

- LoginMemberService 원복
- LoginMemberService -> AuthenticationUserService class 이름 수정
- Override 처리

hjhello423 committed:
refactor: 인자 및 변수명 수정, 기타 리팩토링

- 인자, 변수명 수정
- Override 어노테이션
- 줄맞춤

hjhello423 committed:
refactor: 후처리 메서드 이름 수정

hjhello423 committed:
refactor: 후처리 메서드 이름 수정
